### PR TITLE
Add types spec

### PR DESCRIPTION
**Pull request by chabroA**
Add specification for types based on https://github.com/LedgerHQ/live-app-sdk/blob/main/docs/reference/modules.md

It's a good first step for now but should be refined after first iteration of client, api and common packages implementations to properly organize and define typed that are common to the client and api internal workings, or client or api specific (and exposed to the end-user).

---

Some thoughts on current types:

It looks like some types are not related to the specification (are more implementation specific), like `EcdsaSignature`, `ExchangeDeviceTxId` or `ExchangePayload`.

Also, some types are actually not used anywhere today (like `ApplicationDetails` or `DeviceInfo`).

Some types are also Ledger Live specific like `RawSignedTransaction`.

It feels like all these types should be either removed from the spec or reworked (like `RawSignedTransaction` that seems quite useless as a type since the object it represents is juste passed through the Client SDK from `signTransaction` to `broadcastSignedTransaction`).

Same general feeling regarding types, some types are JS specific (like `BigNumber` or `Buffer`). We should represent them in a language agnostic way (how? what type should we use?)

What do you think? What's your opinion? How should we go about that?

### CODE REVIEW

File: spec/core/types.md [hunk 1]
```diff
@@ -0,0 +1,466 @@
+# Types
+
+Here we describe the types used in the API.
+
+Some types are defined as _raw_, usually in the form `Raw[TYPE-NAME]`. These types are used for communication between the client and server over JSON-RPC. Each property of a raw type must be a JSON primitive data types as described by the JSON-RPC 2.0 [specification](https://www.jsonrpc.org/specification#parameter_structures) (i.e: Strings, Numbers, Booleans, or Null).
+
```
Comment (Justkant) [inline spec/core/types.md:6]: Should we had a table of content so that we can directly jump to a type ?

**Review by Justkant — COMMENTED** [review]
Quick question, green after that

File: spec/core/types.md (hunk 1 above)
Comment (chabroA) [reply spec/core/types.md:6]: Yes 👍

Comment (chabroA) [reply spec/core/types.md:7]: TOC added in [99f04a4](https://github.com/LedgerHQ/wallet-api/pull/8/commits/99f04a4a4e4fd1f8bb66b2042a63907dda2c52d7)

FYI, I used [this extension](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one#table-of-contents) to automatically and reliably generate it 👍

**Review by Justkant — APPROVED** [review]

**Pull request merged**